### ISSUE TITLE
Profinity filter

### ISSUE THREAD

**Issue by hemant933** [abhisheks008/ML-Crate#545]
### ML-Crate Repository (Proposing new issue)
:red_circle: **Profinity filter** :
:red_circle: ** Aim is to classify whether the used text is abusive or not ** :
:red_circle: **Dataset** :
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Participant ID (If not, then put NA) :
- Approach for this Project :
- What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by divyansh2375** [comment]
i want to work on the issue but i didn't work with any machine learning project before can you please give me some resources so that i can solve this issue?

**Comment by HarshRaj29004** [comment]
Harsh Raj
Git Hub: https://github.com//HarshRaj29004
ID: NA
Participant of Ssoc season 3
Can you please assign this task to me

**Comment by useroutofbound** [comment]
Name : Mitul Agarwal
Discord server name : Kaze
GitHub : https://github.com/useroutofbound
ID : NA

Approach : I'll download dataset from hate-speech-twitter and kaggle , and then build a classification model using various vectorization techniques like tf-idf , CountVectorizor etc and use classical models like logistic regression , svm and ensemble based models  and also neural networks for training . Also provide a table showing other predictive scores on both train and test data . And will also try to implement large language models like bert and mistral , if time permits .

SSOC Participant

**Comment by abhisheks008** [comment]
Need to implement classical models such as,
1. Random Forest
2. Decision Tree
3. Logistic Regression
4. Gradient Boosting
5. XGBoost
6. Lasso
7. Ridge
8. MLP Classifier
9. Support Vector Machine

Along with these models you should implement tf-idf and CountVectorizer for this dataset.

Assigning this issue to you @useroutofbound
